public_html/wp-content/mu-plugins/0-error-handling.php

```diff
@@ -0,0 +1,169 @@
+<?php
+defined( 'WPINC' ) or die();
+
+define( 'ERROR_RATE_LIMITING_DIR', '/tmp/error_limiting' );
+
+if ( is_readable( __DIR__ . '/includes/slack/send.php' ) ) {
+	require_once( __DIR__ . '/includes/slack/send.php' );
+}
+
+set_error_handler( 'send_error_to_slack' );
+register_shutdown_function( 'send_fatal_to_slack' );
+
+/**
+ * Error handler to send errors to Slack.
+ *
+ * Always return false so that default error handling still occurs as well.
+ *
+ * @return bool
+ */
+function send_error_to_slack( $err_no, $err_msg, $file, $line ) {
+	if ( ! defined( 'WORDCAMP_ENVIRONMENT' )
+	     || ( 'production' !==  WORDCAMP_ENVIRONMENT && ! defined( 'SANDBOX_SLACK_USERNAME' ) )
+	) {
+		return false;
+	}
+
+	if ( ! init_error_handling() ) {
+		return false;
+	}
+
+	// Checks to see if the error-throwing expression is prepended with the @ control operator.
+	// See https://secure.php.net/manual/en/function.set-error-handler.php
+	if ( 0 === error_reporting() ) {
+		return false;
+	}
+
+	$error_safelist = [
+		E_ERROR,
+		E_USER_ERROR,
+		E_CORE_ERROR,
+		E_COMPILE_ERROR,
+		E_PARSE,
+		E_NOTICE,
+		E_DEPRECATED,
+		E_WARNING,
+	];
+
+	if ( ! in_array( $err_no, $error_safelist ) ) {
+		return false;
+	}
+
+	// Always use constants in the keys here to avoid path disclosure.
+	$error_ignorelist = [
+		// See https://core.trac.wordpress.org/ticket/29204
+		ABSPATH . 'wp-includes/SimplePie/Registry.php:215' => 'Non-static method WP_Feed_Cache::create() should not be called statically',
+	];
+
+	if ( isset( $error_ignorelist[ "$file:$line" ] ) && false !== strpos( $err_msg, $error_ignorelist[ "$file:$line" ] ) ) {
+		return false;
+	}
+
+	// Max file length for ubuntu system is 255.
+	$err_key = substr( base64_encode("$file-$line-$err_no" ), -254 );
+
+	$error_file = ERROR_RATE_LIMITING_DIR . "/$err_key";
+
+	$text = '';
+
+	$data = array(
+		'last_reported_at' => time(),
+		'error_count'      => 0, // since last reported.
+	);
```

Comment:
low priority, but it'd be nice to remove those blank lines and align the `=` operator.